### ISSUE TITLE
Bower breaking node proxy environment if required into a project

### ISSUE THREAD

**Issue by 9point6** [bower/config#31]
A node application I work on requires the `http_proxy` environment variable to function and since updating our bower dependency to the latest version `1.6.3` the variable is set to undefined.

it appears to be related to the code changed in commit 719ef84, but reversing the change doesn't fix the issue.

the issue occurs where the Set method is being called twice so that the original `restoreFrom` object is overwritten with the intermediate environment where the lower-case `http_proxy` variable has been deleted.

The following code can demonstrate this issue:
```javascript
console.log('Pre-bower env', {
  http_proxy: process.env.http_proxy,
  HTTP_PROXY: process.env.HTTP_PROXY,
  https_proxy: process.env.https_proxy,
  HTTPS_PROXY: process.env.HTTPS_PROXY
});

var bower = require('bower');
bower.commands.install()
  .on('end', function() {
    console.log('Post-Bower env', {
      http_proxy: process.env.http_proxy,
      HTTP_PROXY: process.env.HTTP_PROXY,
      https_proxy: process.env.https_proxy,
      HTTPS_PROXY: process.env.HTTPS_PROXY
    });
  });
```

It looks like a check to ensure that `restoreFrom` doesn't get overwritten if it already exists would be a simple fix here, but I'm unsure if there is another issue where this shouldn't be called twice in the first place

**Comment by paulohp** [comment]
@9point6 I just tested here and worked like a glance. Can you post a step by step test case? thanks!

**Comment by 9point6** [comment]
@paulohp Hi, thanks for the response!

I've just verified the code sample I posted previously still breaks with bower@1.6.6

```
Pre-bower env { http_proxy: 'http://proxy.example.com:80',
  HTTP_PROXY: 'http://proxy.example.com:80',
  https_proxy: 'http://proxy.example.com:80',
  HTTPS_PROXY: 'http://proxy.example.com:80' }
Post-Bower env { http_proxy: 'undefined',
  HTTP_PROXY: 'http://proxy.example.com:80',
  https_proxy: 'undefined',
  HTTPS_PROXY: 'http://proxy.example.com:80' }
```

**Comment by paulohp** [comment]
Oh now I see! I'll start working on it soon.

**Comment by paulohp** [comment]
Thanks @9point6!

**Comment by 9point6** [comment]
No worries! :+1:

**Comment by sheerun** [comment]
I fixed it in 1.2.3
